CMakeLists.txt

```diff
@@ -18,6 +18,29 @@ option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_GO_SHLIB "Build Go shared library." OFF)
 option(BUILD_DOCS "Build doxygen documentation (if doxygen is available)." ON)
 
+if(CMAKE_CROSSCOMPILING)
+  set(BOARD_NAME "" CACHE STRING "Enter the board name to compile mlpack for.")
+  set(GFORTRAN_LIBRARY "" CACHE STRING "Path for gfortran library.")
+  set(GCC_VERSION ""  CACHE STRING "")
+endif()
+
+if (CMAKE_CROSSCOMPILING AND (BOARD_NAME STREQUAL ""))
+  message(FATAL_ERROR "Please enter the board name")
```

Comment:
Could it be better to continue anyway here?  CMAKE_CROSSCOMPILING is standard CMake support, so it seems to me that even if a specific board configuration is not specified, if they still specify all the necessary CMake variables it should still work.

Reply:
Yeah, Agreed, the board will only specify the flags, therefore, it is not necessary neither important. 
Of course, it is better to specify the board, but not obligatory.
I will refactor this part